packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx

```diff
@@ -18,10 +18,13 @@ export function UploadPreviewer({
   isLoading,
   isSuccessful,
   hasMaxFilesError,
+  maxFiles,
   onClear,
   onFileClick,
 }: PreviewerProps): JSX.Element {
-  const headingMaxFiles = translate('Over Max files');
+  const headingMaxFiles = `${translate('Over')} ${maxFiles} ${translate(
```

Comment:
I think this error message could be improved. How about something like:
`Cannot choose more than ${maxFiles}`

Reply:
+1, as is translating it to other languages will lead to awkward phrasings.

Reply:
Ooh, good call @ioanabrooks on the translations aspect

Reply:
+1

Reply:
[613ecbb](https://github.com/aws-amplify/amplify-ui/pull/3344/commits/613ecbbec83882cbffd7e56b105c839097dc47d1)

Reply:
I think the wording could be further improved once we have better support for translations with dynamic string, with something like

```
Cannot choose more than ${maxFiles} files
```

I think that looks better in English at least. Not sure if we want to do this now or later though, since our current support dynamic phrases are bad. What do you think? @ioanabrooks @ErikCH @reesscot

Reply:
This feels like something that can be looked at again, after we get translations in. I think we don't want to add the `files` to the end because of the translations.

Reply:
Understood, let's discuss in future translations pr then.